tests/common/plugins/sanity_check/checks.py

```diff
@@ -100,41 +100,56 @@ def check_interfaces(dut):
 def check_bgp_status(dut):
 
     def _check_bgp_status_helper():
-        neighbors.clear()
-        neighbors.update(dut.get_bgp_neighbors())
-        if neighbors:
-            down_neighbors = [k for k, v in neighbors.items()
-                              if v['state'] != 'established']
-            if down_neighbors:
-                check_result['down_neighbors'] = down_neighbors
-                check_result['failed'] = True
+        asic_check_results = []
+        bgp_facts = dut.bgp_facts(asic_index='all')
+        for asic_index, a_asic_facts in enumerate(bgp_facts):
+            asic_check_results.append(False)
+            a_asic_neighbors = a_asic_facts['ansible_facts']['bgp_neighbors']
+            if a_asic_neighbors:
+                down_neighbors = [k for k, v in a_asic_neighbors.items()
+                                  if v['state'] != 'established']
+                if down_neighbors:
+                    if dut.facts['num_asic'] == 1:
+                        check_result['bgp'] = {'down_neighbors' : down_neighbors }
+                    else:
+                        check_result['bgp' + str(asic_index)] = {'down_neighbors' : down_neighbors }
+                    asic_check_results[asic_index] = True
+                else:
+                    asic_check_results[asic_index] = False
+                    if dut.facts['num_asic'] == 1:
+                        if 'bgp' in check_result:
+                            check_result['bgp'].pop('down_neighbors', None)
+                    else:
+                        if 'bgp' + str(asic_index) in check_result:
+                            check_result['bgp' + str(asic_index)].pop('down_neighbors', None)
             else:
-                check_result['failed'] = False
-                check_result.pop('down_neighbors', None)
-        else:
+                asic_check_results[asic_index] = True
```

Comment:
Can you use a variable like `asic_check_result` to hold the result of checking current asic?
Initially it has value `False`. Then update its value according to checking result of the current asic. In the end of each loop, append the variable value to list `asic_check_results`. Then inside the loop, it would be unnecessary to retrieve its value using indexing like `asic_check_results[asic_index]`. IMO, it is a little bit confusing.